Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -149,10 +153,90 @@ private static IEnumerable<SnippetDef> Snippets(SourceFileContext ctx, ServiceDe
         private SnippetDef Snippet { get; }
         public string SnippetMethodName => Snippet.SnippetMethodName;
 
-        public (CompilationUnitSyntax, SnippetMetadata) Generate(SourceFileContext ctx)
+        public (string, SnippetMetadata) Generate(SourceFileContext ctx)
         {
             var (snippetCode, snippetMetadata) = Snippet.FullSnippet(ctx);
-            return (ctx.CreateCompilationUnit(snippetCode), snippetMetadata);
+            // We know need to format the code to make sure that we extract segment line numbers correctly.
+            var formattedCode = CodeFormatter.Format(ctx.CreateCompilationUnit(snippetCode));
+
+            SnippetSegmentFinder.UpdateMetadataWithSegmentInfo(formattedCode, snippetMetadata);
+
+            // Let's return the string so that the code is not formatted or otherwise changed further.
+            return (formattedCode.ToFullString(), snippetMetadata);
+        }
+
+        private class SnippetSegmentFinder : CSharpSyntaxWalker
```

Comment:
Add a comment to this? I can sort of work out the point, but it would be good to have it written out.